### ISSUE TITLE
puller: Moving file to directory of same name causes sync error

### ISSUE THREAD

**Issue by whorfin** [syncthing/syncthing#2767]
Repro case:
A/fubar.pdf/fubar.pdf                 [the PDF file fubar.pdf within the directory named "A/fubar.pdf"]

in the directory A:
mv fubar.pdf fubar.1
mv fubar.1/fubar.pdf fubar.pdf
rmdir fubar.1

This results in 
A/fubar.pdf                               [the PDF file fubar.pdf within the directory named "A"]

Clients fail to sync with messages about being unable to delete "fubar.pdf"

**Comment by AudriusButkevicius** [comment]
Can't repro.

**Comment by AudriusButkevicius** [comment]
Steps, with rescans in between:
```
root@audrius:~/Sync# cd A
root@audrius:~/Sync/A# echo test >foobar.pdf
root@audrius:~/Sync/A# mkdir foobar.1
root@audrius:~/Sync/A# mv foobar.pdf foobar.1/
root@audrius:~/Sync/A# mv foobar.1/foobar.pdf foobar.pdf; rmdir foobar.1/
```

**Comment by whorfin** [comment]
The steps above don't create the situation fully.
Let me suggest instead [full repro now, instead of assuming a working pdf file]

user@machine:~/Sync$ mkdir A
user@machine:~/Sync$ cd A
user@machine:~/Sync$ mkdir foobar.pdf
user@machine:~/Sync$ echo test > foobar.pdf/foobar.pdf
make sure a sync has been done, and confirm on a syncing second machine that the sync is done
and that "A/foobar.pdf/foobar.pdf" exists
user@machine:~/Sync$ mv fubar.pdf fubar.1
user@machine:~/Sync$ mv fubar.1/'fubar.pdf fubar.pdf
user@machine:~/Sync$ rmdir fubar.1
sync, and observe an error on the syncing second machine about "A/fubar.pdf"

**Comment by calmh** [comment]
Fairly sure this is one of the myriad corner cases that will be resolved by #2540 - anyway I'm not going to try to fix it until that's integrated. :)

**Comment by imsodin** [comment]
I can't reproduce anymore, maybe fixed by #4509.
